reports/tests.py

```diff
@@ -528,3 +533,161 @@ def _generate_invoice_date(year=datetime.now().year, month=1, this_month=False):
             date = datetime(year, month, 1)
 
         return timezone.make_aware(date)
+
+
+class ReportTaskTestCase(TestCase):
+    """
+    Tests for reports tasks execution
+    """
+
+    def create_user_with_trial_instance(self, key=''):
+        """
+        Returns a trial instance for a user
+
+        :param key: String to append to properties.
+        :return: A trial instance
+        """
+        user, _ = get_user_model().objects.get_or_create(
+            username=key + 'test', email=key + 'test@example.com'
+        )
+        instance = OpenEdXInstanceFactory(
+            successfully_provisioned=True, betatestapplication=BetaTestApplication()
+        )
+        make_test_appserver(instance=instance, is_active=True)
+        BetaTestApplication.objects.create(
+            user=user,
+            public_contact_email=key + 'publicemail@example.com',
+            subdomain=key + 'betatestdomain',
+            instance_name=instance.name,
+            status=BetaTestApplication.ACCEPTED,
+            instance=instance
+        )
+
+        return instance
+
+    def test_send_trial_instances_report_failure(self):
+        """
+        Tests that if send_trial_instances_report fails because there are no matching instances
+        the recipient gets notified regardless.
+        """
+        send_trial_instances_report(recipients=['to@example.com'])
+        sent_mail = mail.outbox[0]
+        self.assertIn(
+            'Unable to generate a Trial Instances Report due to failure of `instance_statistics_csv`',
+            sent_mail.body
+        )
+        self.assertEqual(
+            [],
+            sent_mail.attachments
+        )
+        self.assertEqual(
+            ['to@example.com'],
+            sent_mail.to
+        )
+        self.assertEqual(len(mail.outbox), 1)
+
+    def test_send_trial_instances_report(self):
+        """
+        Tests that if send_trial_instances_report succeeds, the recipient receives
+        a correct report with only instances created last month.
+        """
+        date1 = "2020-01-01"
+        date2 = "2020-02-01"
+        date3 = "2020-03-01"
+
+        with freeze_time(date1):
+            instance1 = self.create_user_with_trial_instance(key='1')
```

Comment:
@spokerman12, the key doesn't matter as it is not used for any of the checks below. So can't the `create_user_with_trial_instance()` method be a factory method that accepts a count and returns the `count` number of instances, which can be assigned like

```python
instance1, instance2, instance3 = self.user_trial_instance_factory(count=3)
```
?

Reply:
True. Modified as requested.